code/modules/projectiles/guns/throw/pielauncher.dm

```diff
@@ -1,6 +1,7 @@
 /obj/item/gun/throw/piecannon
 	name = "pie cannon"
 	desc = "A projectile weapon that fires pies."
+	icon = 'icons/obj/guns/projectile.dmi'
```

Comment:
ditto.
Generally you have a lot of unnecessary overrides in this pull request. Please take a look at all of them and determine whether they are needed or not. If they're not needed, removed them.